### ISSUE TITLE
Win7: Cyrillic input with shift partially does not work

### ISSUE THREAD

**Issue by MVKozlov** [PowerShell/PSReadLine#798]
Environment data
----------------
```powershell
PS version: 5.1.14409.1012
PSReadline version: 2.0.0-beta3
os: 6.1.7601.17514 (win7sp1_rtm.101119-1850)
PS file version: 10.0.14409.1005 (rs1_srvoob.161208-1155)
```
Steps to reproduce or exception report
--------------------------------------
Installed latest PSReadline under PSv5.1 / Win7

switch keyboard to russian input, type:
- without shift - `йцукенгшщзхъ`
- should be (without shift but with caps) -  `ЙЦУКЕНГШЩЗХЪ`
- with shift only - `ЦУШЩХЪ`
- with caps + shift - `цукшщхъ`

and many other symbols got eaten too

**Comment by MVKozlov** [comment]
Something breaks between beta1 and beta2

**Comment by lzybkr** [comment]
Are you changing the layout after starting PowerShell? If so, this might be hard to fix, the api I rely on the determine if <kbd>Shift</kbd> is necessary does not honor keyboard layout changes.

**Comment by MVKozlov** [comment]
Yes, I start PS and press <kbd>shortcut</kbd> to change layout. The same version (Beta3) works ok on Linux but does not work on Windows (both 7 and 10)

Moreover, if I change default layout to Russian, (I start PSs with RUS layout) I can't get `any` russian upper case symbols with <kbd>shift</kbd> (at least qwerty line) and even when I switch to English I can't type `any` English symbols with <kbd>shift</kbd> as well!

Seems there should be another `PlatformWindows.cs` override

**Comment by ForNeVeR** [comment]
The impact of this issue is very high, because multi-layout users switch their layouts literally all the time. I have some file names and content I want to enter into my PowerShell session in my native layout, and so I need to change my layout from time to time without restarting the whole PowerShell session, that's a disaster.

@lzybkr, how could we help you with the issue? Could you please share any thoughts on that or show where I could find the code of that problematic API and PSReadLine's usage of that API? I'll be glad to help with the fix.

**Comment by ForNeVeR** [comment]
Just wanted to report that it became worse on the latest nightly build at that point: no matter which layout is active when I start PowerShell 6.1 with PSReadLine, I cannot enter any Cyrillic letters at all: some of the keys result in `?` being added to the input and some of them don't work at all.

Will check the build from #822 after it's finished on CI though!

**Comment by lzybkr** [comment]
#822 fixes just the problem initially reported here.

The rendering of `?` (and it's just a rendering problem, the characters are correct if executed) was introduced by #771, the workaround for now is to run:

```PowerShell
[Console]::OutputEncoding = [System.Text.Encoding]::UTF8
```

There is one additional problem if you start PowerShell with the Cyrillic keyboard - PSReadLine will fail to load, and the only workaround right now is to start PowerShell with a different keyboard.

**Comment by ForNeVeR** [comment]
> There is one additional problem if you start PowerShell with the Cyrillic keyboard - PSReadLine will fail to load, and the only workaround right now is to start PowerShell with a different keyboard.

Well, honestly I cannot reproduce that on the latest build (e.g. if I install build from #822). I was able to reproduce that on beta3 though.

Thanks for the tip about `[Console]::OutputEncoding`, it helped!

**Comment by iSazonov** [comment]
@SteveL-MSFT These is very annoying. Can you share information when fixes is expected for Windows?

**Comment by SteveL-MSFT** [comment]
@iSazonov the fix for PSReadLine not starting on non-en-US keyboard layouts is already checked into latest Win10 and should be flighting on latest Insider builds

**Comment by iSazonov** [comment]
@SteveL-MSFT Thanks! Will we get this on WindowsUpdate? For 1809 and Windows 10 Ent 2019 at February or March?

**Comment by SteveL-MSFT** [comment]
@iSazonov for older builds, I think we just need to get another release of PSReadLine to PSGallery

**Comment by iSazonov** [comment]
@SteveL-MSFT But it is enterprise LTS versions.

**Comment by megapro17** [comment]
I'm on 2004 build and still not fixed, PSReadLine version is still 2.0.0
I had to update it manually, and bug is gone
Run this line in PowerShell from administrator:
`Install-Module PSReadLine -force`
